zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -504,60 +536,56 @@ impl<F: Field> SignVerifyChip<F> {
 
         let pk_le = pk_bytes_le(&sign_data.pk);
         let pk_be = pk_bytes_swap_endianness(&pk_le);
-        let pk_hash = (!padding)
-            .then(|| keccak256(&pk_be))
-            .unwrap_or_default()
-            .map(|byte| Value::known(F::from(byte as u64)));
-        let pk_hash_hi = pk_hash[..12].to_vec();
+        let mut pk_hash = (!padding).then(|| keccak256(&pk_be)).unwrap_or_default();
+
         // Ref. spec SignVerifyChip 2. Verify that the first 20 bytes of the
         // pub_key_hash equal the address
-        let (address, pk_hash_lo) = {
+        let address = {
```

Comment:
Let me know if I missed something and you think I'm wrong!  I'm trying to prepare a negative test to validate the missing constraint.

Reply:
I added your change (as following) in `tx_circuit_1tx_1max_tx`, but it's strang that this test failed on my side.  However, I fixed as your suggested in [11db585](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1418/commits/11db585e0bae6b8da05c3575897b13389d9d7d8a)
``` diff
-    let tx: Transaction = mock::CORRECT_MOCK_TXS[0].clone().into();
+    let mut tx: Transaction = mock::CORRECT_MOCK_TXS[0].clone().into();
+    tx.from = address!("0101010101010101010101010101010101010101");
```

Reply:
I tested the current implementation and the tests are not passing (I believe it may be related to endianess).  Nevertheless I think the issue is not resolved: Now you're using the variable `address_word` to calculate the `pk_hash`.  But `address_word` is just a value, it's not any expression built from cells related to the assigned address bytes.

I've made a commit here resolving this, please take a look and if you find it OK you can add it to this PR https://github.com/privacy-scaling-explorations/zkevm-circuits/commit/917963a0a39068f2bb3a8df42e257badcfed5e57

Reply:
Thanks! have merged your commit.